### ISSUE TITLE
Rendering issue of code span and header with Advanced extensions

### ISSUE THREAD

**Issue by xoofx** [lunet-io/markdig#16]
From [Markdown Editor issue 4](https://github.com/madskristensen/MarkdownEditor/issues/4)

With advanced extensions, the following markdown is not rendered properly:

See [example on babelmark](https://babelmark.github.io/?text=%60https%3A%2F%2F%7Bdomain%7D%2Fcallbacks%60%0A%23%23%23%23+HEADING%0AParagraph)

```markdown
`https://{domain}/callbacks`
#### HEADING
Paragraph
```
outputs the invalid html:

```html
<p>
  <code>https://{domain}/callbacks</code>
</p>
<h4 id="section" domain>
</h4>
<p>
  Paragraph
</p>
```

**Comment by madskristensen** [comment]
I can repro with version 0.5.5

**Comment by xoofx** [comment]
This should be fixed by cab3365 part of 0.5.7 version
